### PR TITLE
Falconctl Ansible module support (WIP)

### PR DESCRIPTION
**Pull request by carlosmmatos**
This is a complete Ansible Module for replicating the falconctl cli behavior as best as possible within the current cli limitations. This module consists of 3 parts:

1. `falconctl_utils.py` is a module_utils file that has shared functions to be used across the other falcon modules
2. `falconctl.py` is the main module used to replicate falconctl cli behavior*
3. `falconctl_info.py` is an info module used to get values from the falconctl. You can pass either a list of interested values, or you can simply return all values associated with the GET operation in the cli.

### TODO:

- [x] Clean up class methods using best practices|pep8 formatting
- [x] Clean up/fix all docstrings for both modules (this is the DOCUMENTATION/EXAMPLES/RETURNS sections)
- [x] Create sanitization function(s) for check_mode. Need to deal with edge cases - like cid, metadata-query, etc
- [ ] Double check documentation fragments
- [ ] Create a test plan of different scenarios to ensure we didn't miss any other edge cases. Then run through the test plan.

### CODE REVIEW

File: plugins/modules/falconctl.py [hunk 1]
```diff
@@ -132,32 +146,47 @@ def add_args(self, state):
                     key = k.replace("_", "-")
                     if state == "present":
                         args.append("--%s=%s" %
-                                    (key, self.params[k]))
+                                    (key, self.__list_to_string(self.params[k])))
```
Comment (redhatrises) [inline plugins/modules/falconctl.py:149]: I am included to change all types of `list` to `str`. Thoughts?

Comment (carlosmmatos) [reply plugins/modules/falconctl.py:149]: I'm not opposed. Either way, we'll still need an extra step of input validation IMO

File: plugins/modules/falconctl.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+    feature:
+      description:
+        - Configure the Falcon sensor feature flags.
+      type: list
+      elements: str
+      choices: [ none, enableLog, disableLogBuffer, disableOsfm, emulateUpdate ]
+    metadata_query:
+      description:
+        - Configure the Falcon sensor cloud provider metadata query flags.
+      type: str
```
Comment (carlosmmatos) [inline plugins/modules/falconctl.py:70]: @redhatrises @isimluk Can you guys provide better verbiage for this? Also, please double check the other descriptions and make sure they a) make sense and b) are correct lol..

**Comment by redhatrises** [comment]
Per conversation with Carlos, I am going to merge this with it being this large already. We can make the necessary changes from here as well as make the PRs more digestable with changes.
